packages/http-client-java/generator/http-client-generator/src/main/java/com/microsoft/typespec/http/client/generator/template/TypeSpecEnumTemplate.java

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.typespec.http.client.generator.template;
+
+import com.azure.core.util.CoreUtils;
+import com.microsoft.typespec.http.client.generator.core.extension.plugin.JavaSettings;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ClassType;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.ClientEnumValue;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.EnumType;
+import com.microsoft.typespec.http.client.generator.core.model.clientmodel.IType;
+import com.microsoft.typespec.http.client.generator.core.model.javamodel.JavaFile;
+import com.microsoft.typespec.http.client.generator.core.model.javamodel.JavaModifier;
+import com.microsoft.typespec.http.client.generator.core.model.javamodel.JavaVisibility;
+import com.microsoft.typespec.http.client.generator.core.template.EnumTemplate;
+import com.microsoft.typespec.http.client.generator.core.util.CodeNamer;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * TypeSpec implementation for EnumTemplate.
+ */
+public class TypeSpecEnumTemplate extends EnumTemplate {
+    private static final TypeSpecEnumTemplate INSTANCE = new TypeSpecEnumTemplate();
+
+    public static TypeSpecEnumTemplate getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    protected void writeBrandedExpandableEnum(EnumType enumType, JavaFile javaFile, JavaSettings settings) {
```

Comment:
Unbranded got the ExpandableEnum too. I assume we could share most of this code?
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/clientcore/core/src/main/java/io/clientcore/core/util/ExpandableEnum.java

Reply:
I didn't touch unbranded. 

Actually there seems to be an issue for unbranded non-string expandable enum. Let me combine the implementation.

Reply:
You can do a follow-up PR if you want.

I just think they should be same on TypeSpec branded/unbranded for non-string (but the string case it may be different. unbranded case may also generate a class based on ExpandableEnum).

Reply:
I've updated follow up tasks in this issue: https://github.com/Azure/autorest.java/issues/2841